pkg/aws/machinesets.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/submariner-io/admiral/pkg/resource"
+	"github.com/submariner-io/admiral/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/dynamic"
+)
+
+type machineSetConfig struct {
+	AZ            string
+	AMIId         string
+	InfraID       string
+	InstanceType  string
+	Region        string
+	SecurityGroup string
+	PublicSubnet  string
+}
+
+func (ac *awsCloud) findAMIID(vpcID string) (string, error) {
+	result, err := ac.client.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			ec2Filter("vpc-id", vpcID),
+			ac.filterByName("{infraID}-worker*"),
+			ac.filterByCurrentCluster(),
+		},
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(result.Reservations) == 0 {
+		return "", newNotFoundError("reservations")
+	}
+	if len(result.Reservations[0].Instances) == 0 {
+		return "", newNotFoundError("worker instances")
+	}
+	return *result.Reservations[0].Instances[0].ImageId, nil
```

Comment:
Do we always expect `ImageId` be non-nil?

Reply:
Yes otherwise we cant provision.

For the sake of clarity I'll add an explicit check for it.